js/src/ar.js

```diff
@@ -1,203 +1,106 @@
+    const sessionInit = {
+        optionalFeatures: ["dom-overlay", "hit-test"],
+        domOverlay: { root: closeOverlay }
+    };
 
-        button.style.display = '';
+    showCloseOverlay(true);
 
-        button.style.cursor = 'auto';
-        button.style.left = 'calc(50% - 75px)';
-        button.style.width = '150px';
+    const session = await navigator.xr.requestSession("immersive-ar", sessionInit);
+    renderer.xr.setReferenceSpaceType("local");
+    renderer.xr.setSession(session);
+    const { x, y } = renderer.getSize();
+    lastWidth = x;
+    lastHeight = y;
+    renderer.setSize(window.innerWidth, window.innerHeight);
 
-        button.onmouseenter = null;
-        button.onmouseleave = null;
+    lastXrTransform = null;
 
-        button.onclick = null;
+    renderer.setAnimationLoop(
+        await extractHitTransform(
+            renderer,
+            renderFn,
+            (xrTransform) => { lastXrTransform = xrTransform; }));
 
-    }
-
-    function showARNotSupported() {
-        console.log('ar not supported');
-        disableButton();
-
-        button.textContent = 'AR NOT SUPPORTED';
-
-    }
-
-    function stylizeElement( element ) {
-
-        element.style.position = 'absolute';
-        element.style.bottom = '20px';
-        element.style.padding = '12px 6px';
-        element.style.border = '1px solid #fff';
-        element.style.borderRadius = '4px';
-        element.style.background = 'rgba(0,0,0,0.1)';
-        element.style.color = '#fff';
-        element.style.font = 'normal 13px sans-serif';
-        element.style.textAlign = 'center';
-        element.style.opacity = '0.5';
-        element.style.outline = 'none';
-        element.style.zIndex = '999';
+    setOnClose(onStop);
+}
 
+export function stop(renderer) {
+    if (renderer.xr.getSession() === null) {
+        return;
     }
-    console.log('xr:', 'xr' in navigator);
-    if ( 'xr' in navigator ) {
-
-        button.id = 'ARButton';
-        // button.style.display = 'none';
-
-        stylizeElement( button );
-
-        navigator.xr.isSessionSupported( 'immersive-ar' ).then( function ( supported ) {
-            console.log('supported!:', supported)
-            console.log('context', renderer.getContext());
-            supported ? showStartAR() : showARNotSupported();
-
-        } ).catch( showARNotSupported );
-
-        return button;
-
-    } else {
-
-        const message = document.createElement( 'a' );
-
-        if ( window.isSecureContext === false ) {
-
-            message.href = document.location.href.replace( /^http:/, 'https:' );
-            message.innerHTML = 'WEBXR NEEDS HTTPS'; // TODO Improve message
-
-        } else {
-
-            message.href = 'https://immersiveweb.dev/';
-            message.innerHTML = 'WEBXR NOT AVAILABLE';
+    renderer.xr.getSession().end();
+    renderer.xr.setSession(null);
+    lastXrTransform = null;
+    showCloseOverlay(false);
+    renderer.setAnimationLoop(() => null);
```

Comment:
```suggestion
    renderer.setAnimationLoop(null);
```
Otherwise we still render continuous after stopping.